src/azure-cli/azure/cli/command_modules/synapse/constant.py

```diff
@@ -62,3 +64,17 @@ class SqlPoolConnectionClientAuthenticationType(str, Enum):
     SqlPassword = 'SqlPassword'
     ActiveDirectoryPassword = 'ADPassword'
     ActiveDirectoryIntegrated = 'ADIntegrated'
+
+
+class PrincipalType(str, Enum):
+    user = "User"
+    group = "Group"
+    service_principal = "ServicePrincipal"
+
+
+class ItemType(str, Enum):
+    bigDataPools = "bigDataPools"
+    scopePools = "scopePools"
+    integrationRuntimes = "integrationRuntimes"
+    credentials = "credentials"
+    linkedServices = "linkedServices"
```

Comment:
why not define in SDK?

Reply:
@Juliehzl  I think this not belong to SDK's scope, we define these item type for CLI using friendly. User can pass item type and CLI code will use this parameter to generate `scope` which is defined in SDK and received by service. 

`PrincipalType` and `ItemType` 's purpose is only to limit the user's input

@idear1203  Do we have a plan to move it to SDK?

Reply:
Currently we don't have such definitions in SDK. This is a good suggestion.

Reply:
I see.